utils.js

```diff
@@ -48,11 +48,15 @@ module.exports = {
    * @param {object} reg
    */
   getHeaderOptions: function(req){
-    if(req && req.req && req.req._headers){
-      return req.req._headers;
+    //I have to remove the User-Agent header ever since superagent 1.7.0
```

Comment:
I did this in order to maintain backwards compatibility with the headers output from versions of superagent before `1.7.0`. I see what you're saying and I'm open to other solutions. It was a long time ago that I made this change, so I don't remember all the reasons I did/didn't have, but I'm willing to consider better options.